bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -1685,17 +1687,29 @@ private static boolean checkObjectEquivalency(Object sourceVal, Type sourceType,
 
         Map<String, Field> targetFields = targetType.getFields();
         Map<String, Field> sourceFields = sourceObjectType.getFields();
-        MethodType[] targetFuncs = targetType.getMethods();
-        MethodType[] sourceFuncs = sourceObjectType.getMethods();
+        List<MethodType> targetFuncs = new ArrayList<>(Arrays.asList(targetType.getMethods()));
+        List<MethodType> sourceFuncs = new ArrayList<>(Arrays.asList(sourceObjectType.getMethods()));
+        
+        if (targetType.getTag() == TypeTags.SERVICE_TAG || 
+                (targetType.flags & SymbolFlags.CLIENT) == SymbolFlags.CLIENT) {
+            BNetworkObjectType targetNetworkObj = (BNetworkObjectType) targetType;
+            Collections.addAll(targetFuncs, targetNetworkObj.getResourceMethods());
+        }
+
+        if (sourceType.getTag() == TypeTags.SERVICE_TAG || 
+                (sourceObjectType.flags & SymbolFlags.CLIENT) == SymbolFlags.CLIENT) {
+            BNetworkObjectType sourceNetworkObj = (BNetworkObjectType) sourceObjectType;
+            Collections.addAll(sourceFuncs, sourceNetworkObj.getResourceMethods());
+        }
```

Comment:
I think that it will be clearer if we can write a small method and call two times with passing arguments appropriately.

Reply:
Updated in [c7f4e52](https://github.com/ballerina-platform/ballerina-lang/pull/37494/commits/c7f4e52017b2fe3dbcede63d7fbe387cbeca5ea8)